pbr/scene/misc_robot.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/local/blender -P
+
+import os
+import bpy
+import json
+import re
+from math import radians
+from random import triangular, randint
+
+from config import blend_config as blend_cfg
+from config import scene_config as scene_cfg
+
+from scene.blender_object import BlenderObject
+
+
+class MiscRobot(BlenderObject):
+    def __init__(self, name, class_index, robot_info):
+        self.mat = {}
+        self.sc_plane = None
+        self.robot = None
+        self.robot_parts = None
+        self.pass_index = class_index
+        self.objs = {}
+        self.obj = None
+        self.name = name
+        self.colour = randint(0, 1)  # 1. white or 0. black
+        self.height = 0
+        self.construct(robot_info)   
+
+    # Setup robot object
+    def construct(self, robot_info):
+        robot_mesh = None
+        robot_obj = {}
+
+        self.robot = robot_info
+
+        mesh_path = self.robot["mesh_path"]
+
+        self.height = self.robot["height"]
+
+        with open(self.robot["kinematics_path"], "r") as file:
+            self.robot_parts = json.loads(file.read())
+
+        bpy.ops.import_scene.fbx(
+            filepath=self.robot["mesh_path"], axis_forward="X", axis_up="Z"
+        )
+
+        # Pure black is too dark
+        if (self.colour == 0):
+            self.colour = 0.1
+
+        for p in self.robot_parts.keys():
+            obj = bpy.data.objects[p]
+
+            obj.data.materials.clear()
+
+            if obj.parent == None:
+                self.obj = obj
+
+            obj.name = "{}_{}".format(self.name, p)
+            self.objs.update({obj.name: obj})
+
+            obj.pass_index = self.pass_index
+
+            # Configure robot to have correct pass index
+            obj.pass_index = self.pass_index
+
+            # TODO: Make materials not embedded in FBX files
+            # Set material for robot
+            self.mat.update({obj.name: self.set_material(obj, self.name + "_tex")})
+            obj.data.materials.append(self.mat[obj.name])
+
+        self.initialise_kinematics()
+
+    def set_material(self, obj, mat_name):
+        l_mat = bpy.data.materials.new(mat_name)
+
+        # Enable use of material nodes
+        l_mat.use_nodes = True
+
+        # Get our node list to construct our material
+        node_list = l_mat.node_tree.nodes
+
+        # Clear node tree of default settings
+        for node in node_list:
+            node_list.remove(node)
+
+        # Construct node tree
```

Comment:
Is code supposed to be under this comment?

Reply:
Nope, and that section is mostly copied from robot.py anyway.